pubpub/platform
# Enable relationships between pubs

## Issue(s) Resolved
https://github.com/pubpub/v7/issues/605

## High-level Explanation of PR
Database schema changes to support relationships between pubs. Although the database changes are in, I've filtered out pub fields that have `isRelated=true` for now, so there should be no noticeable difference in the UI.

## Test Plan

As a user, I want to "connect" two pubs by saying one pub is a contributor of another.

1. Create a pub of type "Submission"
2. Create a new pub type "Author" with any fields (i.e. maybe an email)
3. Create a pub of type "Author" 
4. Directly in your db or via Prisma Studio, add a pub field "Contributes To" with `isRelation=true` and schema type 'String'
5. Add "Contributes To" to the "Author" pub type 
6. Add a PubValue to your "Author" pub of field "Contributes To". You can leave its value null (but you could also imagine giving it a string i.e. "First Author" to denote the contribution type. Give it a `relatedPubId` of your "Submission" pub
7. Make sure the app still works!

Steps 4-6 have to be done directly in the db, as we have purposefully added checks to make sure that these "relation" fields do not show up in the UI yet.

## Screenshots (if applicable)

## Notes
